src/deluge/gui/ui/load/load_song_ui.cpp

```diff
@@ -226,233 +227,258 @@ ActionResult LoadSongUI::buttonAction(deluge::hid::Button b, bool on, bool inCar
 	return ActionResult::DEALT_WITH;
 }
 
+enum class LoadStatus { start, loadingSamples, waitingToEnd, newSongPlaying, finished };
+StorageManager* bdsm{};
 // Before calling this, you must set loadButtonReleased.
-void LoadSongUI::performLoad(StorageManager& bdsm) {
-
-	FileItem* currentFileItem = getCurrentFileItem();
+void LoadSongUI::performLoad(StorageManager& _bdsm) {
+	bdsm = &_bdsm;
+	performLoadFixedSM();
+}
+void LoadSongUI::performLoadFixedSM() {
+	static int32_t count = 0;
+	Song* toDelete = nullptr;
+	static LoadStatus status = LoadStatus::start;
+	if (status == LoadStatus::finished)
+		status = LoadStatus::start;
```

Comment:
```suggestion
	if (status == LoadStatus::finished) {
		status = LoadStatus::start;
	}
```

Style nits aside, should this reset happen in `performLoad` instead with a freeze if we try to start a load while another is ongoing?